pkg/collector/helm3.go

```diff
@@ -43,6 +45,20 @@ func NewHelmV3Collector(opts *HelmV3Opts) (*HelmV3Collector, error) {
 	configDriver := driver.NewConfigMaps(collector.client.ConfigMaps(""))
 	collector.configStore = storage.Init(configDriver)
 
+	if opts.DiscoveryClient == nil {
```

Comment:
seems like we are repeating ourselves alot here

Reply:
I did a bit of refactoring to get rid of a lot of the duped code. Should be much better now.